packages/lesswrong/lib/collections/comments/helpers.ts

```diff
@@ -22,7 +27,13 @@ export async function commentGetPageUrlFromDB(comment: DbComment, isAbsolute = f
     const prefix = isAbsolute ? getSiteUrl().slice(0,-1) : '';
     const tag = await mongoFindOne("Tags", {_id:comment.tagId});
     if (!tag) throw Error(`Unable to find ${taggingNameSetting.get()} for comment: ${comment._id}`)
-    return `${prefix}/${taggingNameIsSet.get() ? taggingNamePluralSetting.get() : 'tag'}/${tag.slug}/discussion#${comment._id}`;
+
+    if (comment.tagCommentType === TagCommentType.Discussion) {
+      return `${prefix}/${taggingNameIsSet.get() ? taggingNamePluralSetting.get() : 'tag'}/${tag.slug}/discussion#${comment._id}`;
+    } else {
+      return `${prefix}${tagGetSubforumUrl(tag)}#${comment._id}`;
```

Comment:
I'm not 100% sure that it will actually scroll the correct comment into view atm, but it does go to the right page, and currently there are few enough comments that replies will tend to be near the bottom anyway. I'll make sure it does work at a later date